HAWQ-1455. Wrong results on CTAS query over catalog

paul-guo- opened the pull request:
This reverts the previous fix for HAWQ-512, however to HAWQ-512, it looks like
we could modify the lock related code following gpdb to do a real fix. Those code
was probably deleted during early hawq development.

ztao1987 commented:
Seems these codes comes from a bunch of bug fix involved from GPDB commit 6b0e52beadd678c5050af9c978c26d171ba86ae0. For this specific bug, I have no doubt it works. But for lock mechanism, gpdb fix involves more than that. Not sure if there is potential issue in lock.

paul-guo- commented:
My guess is that we simplified the lock mechanism, i.e. we removed some
code from gpdb. That patch seems to be the whole gpdb patch against pg for
open source.

2017-05-10 17:14 GMT+08:00 zhenglin tao <notifications@github.com>:

> Seems these codes comes from a bunch of bug fix involved from GPDB commit
> 6b0e52beadd678c5050af9c978c26d171ba86ae0. For this specific bug, I have
> no doubt it works. But for lock mechanism, gpdb fix involves more than
> that. Not sure if there is potential issue in lock.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-hawq/pull/1232#issuecomment-300423845>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHI5jJ1cITueX3x4Z_fYO19GnsvLxfl0ks5r4X_ngaJpZM4NRu03>
> .
>

ztao1987 commented:
Looks good to me. +1

huor commented:
+1 for the fix
